File: src/leetcode/2025/10/Solution26.java

public class Solution26 {
    public static void main(String[] args) {
        int[] nums1 = { 1, 1, 2 };
        int len1 = new Solution26().removeDuplicates(nums1);
        System.out.println(len1); // Output: 2

        int[] nums2 = { 0, 0, 1, 1, 1, 2, 2, 3, 3, 4 };
        int len2 = new Solution26().removeDuplicates(nums2);
        System.out.println(len2); // Output: 5
    }

    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int slow = 0;
        for (int fast = 1; fast < nums.length; fast++) {
            if (nums[fast] != nums[slow]) {
                slow++;
                nums[slow] = nums[fast];
            }
        }
        return slow + 1;
    }
}

File: src/leetcode/2025/10/Solution26.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        if (nums.empty())
            return 0;

        int writeIndex = 1;
        for (int readIndex = 1; readIndex < nums.size(); readIndex++)
        {
            if (nums[readIndex] != nums[readIndex - 1])
            {
                nums[writeIndex] = nums[readIndex];
                writeIndex++;
            }
        }
        return writeIndex;
    }
};

int main()
{
    Solution solution;
    vector<int> nums1 = {1, 1, 2};
    int len1 = solution.removeDuplicates(nums1);
    cout << len1 << endl; // Output: 2

    vector<int> nums2 = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
    int len2 = solution.removeDuplicates(nums2);
    cout << len2 << endl; // Output: 5

    return 0;
}